lib/ContractInteract/PricerRule.js

```diff
@@ -0,0 +1,348 @@
+      organization,
+      eip20Token,
+      bytesBaseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals,
+      tokenRules
+    ];
+    const contract = Contracts.getPricerRule(auxiliaryWeb3);
+
+    return contract.deploy({
+      data: bin,
+      arguments: args
+    });
+  }
+
+  /**
+   * Adds a new price oracle. From address should be only worker address.
+   *
+   * @param {String} priceOracleAddress PriceOracle contract address.
+   * @param {Object} txOptions Tx options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async addPriceOracle(priceOracleAddress, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address ${txOptions.from} in transaction options.`);
+      return Promise.reject(err);
+    }
+    const txObject = await this.addPriceOracleRawTx(priceOracleAddress);
+    return Utils.sendTransaction(txObject, txOptions);
+  }
+
+  /**
+   * Removes price oracle contract address for pay CurrencyCode. From address should be only worker address.
+   *
+   * @param {String} payCurrencyCode QuoteCurrency code. e.g. USD, ETH, BTC.
+   * @param {Object} txOptions Tx options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async removePriceOracle(payCurrencyCode, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address ${txOptions.from} in transaction options.`);
+      return Promise.reject(err);
+    }
+    const txObject = await this.removePriceOracleRawTx(payCurrencyCode);
+    return Utils.sendTransaction(txObject, txOptions);
+  }
+
+  /**
+   * Sets an acceptance margin for the base currency price per pay
+   *         currency. From address should be only worker address.
+   *
+   * @param {String} payCurrencyCode QuoteCurrency code. e.g. USD, ETH, BTC.
+   * @param {String} acceptanceMargin Acceptance margin for the base currency price per pay currency.
+   * @param {Object} txOptions Tx options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async setAcceptanceMargin(payCurrencyCode, acceptanceMargin, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address ${txOptions.from} in transaction options.`);
+      return Promise.reject(err);
+    }
+    let txObject = await this.setAcceptanceMarginRawTx(payCurrencyCode, acceptanceMargin);
```

Comment:
Do we need await here?
 This works fine ? `setAcceptanceMarginRawTx ` doesn't returns promise, I think.

Similar comments at other places.

Reply:
Updated rawTx methods such that they return Promise now. So await is needed.